### PR TITLE
Added a Kafka+SQL storage variant

### PR DESCRIPTION
**Pull request by EricWittmann**

### CODE REVIEW

File: storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+    Integer pollTimeout;
+    
+    @Inject
+    KafkaSqlCoordinator coordinator;
+
+    @Inject
+    KafkaSqlDispatcher dispatcher;
+    
+    private boolean stopped = true;
+    private KafkaProducer<String, JournalRecord> producer;
```
Comment (alesj) [inline storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java:121]: Why not using producer/consumer from Registry's Kafka Util ?

File: storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+        stopped = true;
+    }
+
+    /**
+     * Start the KSQL Kafka consumer thread which is responsible for subscribing to the kafka topic, 
+     * consuming JournalRecord entries found on that topic, and applying those journal entries to
+     * the internal data model.
+     * @param consumer
+     */
+    private void startConsumerThread(final KafkaConsumer<String, JournalRecord> consumer) {
```
Comment (alesj) [inline storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java:150]: Exactly the point in previous comment -- use existing Registry's Kafka Utils,
as then there is no need for this code + it's a lot more robust.

File: storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+        // Configure kafka settings
+        props.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        props.putIfAbsent(ProducerConfig.CLIENT_ID_CONFIG, "Producer-" + topic);
+        props.putIfAbsent(ProducerConfig.ACKS_CONFIG, "all");
+        props.putIfAbsent(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        props.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JournalRecordSerializer.class.getName());
+
+        // Create the Kafka producer
+        KafkaProducer<String, JournalRecord> producer = new KafkaProducer<>(props);
+        return producer;
```
Comment (alesj) [inline storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java:219]: Same for producer, use Utils -- it's resilient to exceptions that can be handled ...

File: storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java (hunk 1 above)
Comment (EricWittmann) [reply storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java:121]: Will do, thanks.

File: storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java (hunk 2 above)
Comment (EricWittmann) [reply storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java:150]: Will do, thanks.

**Pull request closed without merging**